src/makeCalculator.js

```diff
@@ -4,7 +4,37 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    operate(action, value) {
+      action.call(this, value);
+
+      return this;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    add(value) {
+      this.result += value;
+    },
+
+    subtract(value) {
+      this.result -= value;
+    },
+
+    multiply(value) {
+      this.result *= value;
+    },
+
+    divide(value) {
+      this.result = value === 0 ? 0 : this.result / value;
```

Comment:
The divide method correctly checks for division by zero, which is a good practice to avoid runtime errors. However, returning 0 when dividing by zero may not be the best approach as it can be misleading. It might be better to throw an error or return 'undefined' to indicate that the operation is invalid.